doc/nrf-bm/libraries/bm_zms.rst

```diff
@@ -0,0 +1,350 @@
+
+``(5 * ATE_SIZE)``: Reserved ATEs for header and delete items
+
+``DATA_SIZE``: Size of the small data values (range from 1 to 8)
+
+For example for 4 sectors of 1024 bytes, free space for 8-byte length data is :math:`\frac{3 \times 944 \times 8}{16} = 1416 \, \text{ bytes}`.
+
+Large data values
+=================
+
+Large data values ( > 8 bytes) are stored separately at the top of the sector.
+
+In this case, it is hard to estimate the free available space, as it depends on the size of the data.
+However, it can be taken into account that for N bytes of data (N > 8 bytes), an additional 16 bytes of ATE must be added at the bottom of the sector.
+
+For example, for a partition that has 4 sectors of 1024 bytes and for data size of 64 bytes, only 3 sectors are available for writes with a capacity of 944 bytes each.
+Each key-value pair needs an extra 16 bytes for the ATE, which makes it possible to store 11 pairs in each sector (:math:`\frac{944}{80}`).
+Total data that that can be stored in this partition in this case is :math:`11 \times 3 \times 64 = 2112 \text{ bytes}`.
+
+Wear leveling
+*************
+
+This storage system is optimized for devices that do not require an erase.
+
+BM_ZMS uses a cycle count mechanism that avoids emulating erase operations for these devices.
+It also guarantees that every memory location is written to only once for each cycle of a sector write.
+
+The garbage collection operation also reduces the memory cell life expectancy as it performs write operations when moving blocks from one sector to another.
+To prevent the garbage collector from affecting the life expectancy of the device, it is recommended to set the size of the partition appropriately.
+Its size should be the double of the maximum size of data (including headers) that could be written in the storage.
+
+See `Available space for user data <#available-space-for-user-data-key-value-pairs>`_.
+
+Device lifetime calculation
+===========================
+
+Storage devices, whether they are traditional flash or newer technologies like RRAM/MRAM, have a limited life expectancy which is determined by the number of times memory cells can be erased and written to.
+
+Flash devices are erased one page at a time as part of their functional behavior (otherwise memory cells cannot be overwritten), and for storage devices that do not require an erase
+operation, memory cells can be overwritten directly.
+
+The following is a typical scenario that allows to calculate the life expectancy of a device:
+
+Assuming that an 8-byte variable is stored using the same ID but its content changes every minute.
+The partition has 4 sectors with 1024 bytes each.
+Each write of the variable requires 16 bytes of storage.
+As there are 944 bytes available for ATEs for each sector, and because BM_ZMS is a fast-forward storage system, the first location of the first sector is going to be rewritten after :math:`\frac{(944 \times 4)}{16} = 236 \text{ minutes}`.
```

Comment:
```suggestion
As there are 944 bytes available for ATEs for each sector, and because BM_ZMS is a fast-forward storage system, the first location of the first sector is going to be rewritten after :math:`\frac{(944 \times 4)}{16} = 236 \text{minutes}`.
```